homeassistant/components/emulated_hue/__init__.py

```diff
@@ -122,54 +149,10 @@ async def async_setup(hass: HomeAssistant, yaml_config: ConfigType) -> bool:
     HueGroupView(config).register(app, app.router)
     HueFullStateView(config).register(app, app.router)
 
-    listen = create_upnp_datagram_endpoint(
-        config.host_ip_addr,
-        config.upnp_bind_multicast,
-        config.advertise_ip,
-        config.advertise_port or config.listen_port,
-    )
-    protocol: UPNPResponderProtocol | None = None
+    async def _start(event: Event) -> None:
+        """Start the bridge."""
+        await start_emulated_hue_bridge(hass, config, app)
 
-    async def stop_emulated_hue_bridge(event):
-        """Stop the emulated hue bridge."""
-        nonlocal protocol
-        nonlocal site
-        nonlocal runner
-
-        if protocol:
```

Comment:
I refactored this a bit to get rid of the `nonlocal` as mypy couldn't figure it out